### ISSUE TITLE
TraceIdRatioBased default ratio when configured with OTEL_TRACE_SAMPLER

### ISSUE THREAD

**Issue by anuraaga** [open-telemetry/opentelemetry-specification#1310]
Does TraceIdRatioBased have a default ratio or should SDKs throw an error if a ratio isn't provided? This would generally not be an in issue in programatic configuration, but if using environment variables as [here](https://github.com/open-telemetry/opentelemetry-specification/blob/f2869a369ff4b93024e3437837320da76d2c0637/specification/sdk-environment-variables.md#general-sdk-configuration), the sampler name could be set without an arg. I think 1.0 could be a reasonable default, or an error could make sense too.

**Comment by carlosalberto** [comment]
I'd go for 1.0 as well.

cc @yurishkuro

**Comment by Oberon00** [comment]
From my comment https://github.com/open-telemetry/opentelemetry-specification/pull/1190#discussion_r517151004: I think a value less than 1 (100%) makes more sense, as you would usually enable this sampler to reduce load.

**Comment by anuraaga** [comment]
@Oberon00 I see - could go with 0.01 too, but I feel like any default is not applicable to anything more than a slice of use cases. In that case, I'd rather go with an error instead, or if no errors, than log + disable.

**Comment by carlosalberto** [comment]
For reference: the Jaeger Collector [defaults to a very small value](https://www.jaegertracing.io/docs/1.21/sampling/#collector-sampling-configuration) ( less than 1%) when the probabilistic sampler is used and no parameter is specified.

**Comment by yurishkuro** [comment]
We discussed this already somewhere else. The argument for 100% sampling is that when people are trying out the technology they don't constantly run into the issue "where are my traces" because the default sampling is too small. The downside of 100% sampling is obviously perf impact and backend capacity, but strictly speaking it's not much different from standing up a single ES node and pointing 1M clients to it - the need for proper scaling of a distributed backend is generally well understood.

In other words, on Jaeger chat room we've seen plenty of "where are my traces" and almost never "Jaeger is not able to capture all of them". Defaulting to 100% sampling solves the former (although if our default was different we might have heard the latter more often).

**Comment by anuraaga** [comment]
@yurishkuro Well, OTEL itself defaults to 100% sampling. So I can see @Oberon00's point that a lower number as default for ratio sampler could be more natural since the reason to use the ratio sampler is to get less traces than the default. But I can't think of a reason to pick any of 0.1, 0.01, 0.001 it would be arbitrary, so 1.0 does have the nice aspect that it is an unambiguous decision.

**Comment by carlosalberto** [comment]
>  I can't think of a reason to pick any of 0.1, 0.01, 0.001 it would be arbitrary

Yes, indeed. Because of this, and what @yurishkuro mentioned, I'd rather stay on the _safe_ side and go for 1.0 - users who truly want to use this sampler will _have_ to configure it, and that's not entirely bad IMHO.
